packages/contracts/contracts/examples/UnirepVoting.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+import {Unirep} from '.././Unirep.sol';
+
+// Uncomment this line to use console.log
+// import 'hardhat/console.sol';
+import {ReputationVerifierHelper} from '../verifierHelpers/ReputationVerifierHelper.sol';
+
+contract UnirepVoting {
+    Unirep public unirep;
+    ReputationVerifierHelper public helper;
+    uint8 public numOptions;
+    uint8 public numVoters;
+    int[] public scores;
+    mapping(uint256 => uint256) public voted;
+
+    constructor(
+        Unirep _unirep,
+        ReputationVerifierHelper _helper,
+        uint8 _numOptions,
+        uint8 _numVoters,
+        uint48 _epochLength
+    ) {
+        // set unirep address
+        unirep = _unirep;
+
+        // set verifier address
+        helper = _helper;
+
+        numOptions = _numOptions;
+        numVoters = _numVoters;
+
+        scores = new int[](numOptions);
+
+        unirep.attesterSignUp(_epochLength);
+    }
+
+    // sign up users in this app
+    function userSignUp(
+        uint256[] calldata publicSignals,
+        uint256[8] calldata proof
+    ) public {
+        unirep.userSignUp(publicSignals, proof);
+    }
+
+    function vote(
+        uint256 option,
+        uint256[] calldata publicSignals,
+        uint256[8] calldata proof
```

Comment:
I think it can be epoch key proof here?
just to make sure the user has signed up in this app
and we can reveal the epoch key nonce (e.g. `nonce = 0`) to make sure the user won't vote many times

Reply:
hm should user submit 2 proofs? one reputation and one epoch key proof?

Reply:
ah I think it could be `signup` and `joinProject`
and signup can be for hackers and mentors and others
and joinProject is only for hackers
and `signup` can take the signup proof
and `joinProject` can take epoch key proof

Reply:
ok ill try implement this - i had a different version in my implementation!

Reply:
without dataProof, how do you check if a user is part of a project or not?

Reply:
i revised it to be: 

joinProject with epoch key proof
vote takes epoch key proof
claim prize takes reputation proof (im not sure which field is useful here, is it minRep and maxRep? what are they?)

Reply:
> without dataProof, how do you check if a user is part of a project or not?

just use reputation proof
since the user will receive data from every vote
(so we can protect user's privacy, he doesn't have to reveal the project he joined)

> claim prize takes reputation proof (im not sure which field is useful here, is it minRep and maxRep? what are they?)
Yes the `minRep` field
It checks if `data[0]-data[1] >= minRep` or not

https://github.com/Unirep/Unirep/blob/67b2b5989b69c52568dda0b7e478e92ffadb83d8/packages/circuits/circuits/proveReputation.circom#L102